### PR TITLE
Add UpdateAddrs allowing as per zookeeper documentation http://zookeeper...

### PR DESCRIPTION
**Pull request by domwong**
....apache.org/doc/trunk/zookeeperProgrammers.html#ch_zkSessions

### CODE REVIEW

File: zk/structs.go [hunk 1]
```diff
@@ -582,3 +587,35 @@ func responseStructForOp(op int32) interface{} {
...
+	}
+	return false
+}
+
+func (s *serverList) changed(addrs []string) bool {
+	if len(addrs) != len(s.addrs) {
+		return true
+	}
+
+	for _, addr := range addrs {
```
Comment (domwong) [inline zk/structs.go:605]: Doesn't catch the case where we have dupes. e.g.
old addrs = [1234, 5678]
new addrs = [1234, 1234]
returns false

File: zk/structs.go [hunk 2]
```diff
@@ -582,3 +587,35 @@ func responseStructForOp(op int32) interface{} {
...
+	return false
+}
+
+func (s *serverList) next() string {
+	s.index = (s.index + 1) % len(s.addrs)
+	return s.addrs[s.index]
+}
+
+func (s *serverList) hasNext() bool {
+	return len(s.addrs) > 0 && s.index+1 < len(s.addrs)
```
Comment (domwong) [inline zk/structs.go:632]: Does this need to do modulo arithmetic like above?

File: zk/structs.go (hunk 1 above)
Comment (asim) [inline zk/structs.go:605]: Added a reverse check on hosts that catches this. Test passes.

File: zk/structs.go (hunk 2 above)
Comment (asim) [inline zk/structs.go:632]: No this checks whether we are past the bounds of the slice rather than trying to assess whether we are back at the start.

**Pull request merged**